File: src/main/org/nlogo/prim/etc/_randomycor.java
package org.nlogo.prim.etc;

import org.nlogo.nvm.Context;
import org.nlogo.nvm.Reporter;
import org.nlogo.api.Syntax;

public final strictfp class _randomycor extends Reporter {
  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax(Syntax.NumberType());
  }

  @Override
  public Object report(Context context) {
    return report_1(context);
  }

  public double report_1(Context context) {
    double min = world.minPycor() - 0.5;
    double max = world.maxPycor() + 0.5;
    return min + context.job.random.nextDouble() * (max - min);
  }
}

File: src/main/org/nlogo/prim/etc/_randomycor.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Reporter }

class _randomycor extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(Syntax.NumberType)
  override def report(context: Context): java.lang.Double =
    Double.box(report_1(context))
  def report_1(context: Context): Double = {
    val min = world.minPycor - 0.5
    val max = world.maxPycor + 0.5
    min + context.job.random.nextDouble * (max - min)
  }
}
